signing/signing.go

```diff
@@ -232,8 +232,8 @@ func verifyScorecardWorkflow(workflowContent string) error {
 	}
 
 	// Verify that the all scope, if set, isn't write-all.
-	globalPermAll := workflow.Permissions.All
-	if globalPermAll != nil && globalPermAll.Value == "write-all" {
+	globalPerm := workflow.Permissions
```

Comment:
Might need a similar check on line 230. `workflow == nil`

Reply:
and at line 241 which also accesses the `Permissions` field

Reply:
Fixed + added tests